### DISCUSSION TITLE
PO file translations - MultipleObjectsReturned error

### DISCUSSION THREAD

**Discussion by byandrej** [wagtail/wagtail-localize#606] [Q&A] [answered]
I was trying to translate a page using PO editor. But I got error: get() returned more than one String -- it returned 2!
An error was made by one of query selector in wagtail_localize/models.py:

        for index, entry in enumerate(po):
            try:
                string = String.objects.get( …
                    locale_id=self.source.locale_id, data=entry.msgid
                )
               ...
with kwargs {'data': 'Lokacija', 'locale_id': 1}, 
clone:<QuerySet [<String: String object (662)>, <String: String object (663)>

In wagtail_localize_string there were two strings with the same name but one started with a capital letter and the other with a small letter (One is probably title and the other is slug). In DB I found 10 duplicated strings, all of them with same situation.
<html>
<body>

Modify | id | data_hash | data | locale_id
-- | -- | -- | -- | --
uredi | 662 | 331eb28504df5b48966cf4b1ec541ba8 | Lokacija | 1
uredi | 663 | 8d7c8d74ab4f5957ab6a9670d04e6b7c | lokacija | 1

</body>
</html>

Any idea how to solve this?

Django Version: 4.0.6
Python Version: 3.10.4
Wagtail: 2.16.2
wagtail-localize: 1.2.1

**Comment by zerolab** [comment]
@byandrej what database (and version) are you using?

I would expect `String.objects.get(locale_id=1, data="Lokacija")` to return just 662 🤔

**Comment by chris48s** [comment]
> @byandrej what database (and version) are you using?

I'm going to go out on a limb and guess: MySQL with a case-insensitive table collation?

**Reply by zerolab** [reply]
that was my thinking too...
I guess we could change the code to something like:

```python
string = String.objects.filter(
    # ...
    locale_id=self.source.locale_id, data=entry.msgid
).first()
```

to cater for setups like this, even though I would very much push for a case sensitive collation which should be straightforward to do (via `ALTER DATABASE db_name COLLATE collation_name_cs;` https://dev.mysql.com/doc/refman/8.0/en/charset-database.html)

**Reply by byandrej** [reply]
True, I'm using MySQL 5.7 with utf8 case insensitive collection. Better support for utf8 cs was introduced with v8. In v5.7, only option is to change it to utf8_bin, not ideal, but it works.

In the future I suggest that code should be changed to work with _ci db too.  But for now, I'll accept this solution.

**Comment by Nigel2392** [comment, marked as answer]
I have adressed this in the following issue: https://github.com/wagtail/wagtail-localize/issues/758
